content/en/docs/features/encrypted-images.md

```diff
@@ -244,3 +244,51 @@ kubectl logs -f deploy/kbs | grep "$KEY_PATH"
 
 > **Note:** If you're not using KBS deployment using trustee operator additional namespace may be needed `-n coco-tenant`.
 
+## Debugging
+
+The encrypted image feature relies on a cascade of other features as building blocks:
+
+- Image decryption is built on secret retrieval from KBS
+- Secret retrieval is built on remote attestation
+- Remote attestation requires retrieval of hardware evidence from the TEE
+
+All of the above have to work in order to decrypt an encrypted image.
+
+### 0. Launching with an unencrypted image
+
+Launch the same image with unencrypted layers _from the same registry_ to verify that the image itself is not an issue.
+
+### 1. Retrieve hardware evidence from TEE
+
+Launch an unencrypted `library/nginx` deployment named `nginx` with a CoCo runtime class. Issue `kubectl exec deploy/nginx -- curl http://127.0.0.1:8006/aa/evidence\?runtime_data\=xxxx`. This should produce a real hardware evidence rendered as JSON. Something like `{"svn":"1","report_data":"eHh4eA=="}` is not real hardware evidence, it's a dummy response from the sample attester. Real TEE evidence should be more verbose and contain certificates and measurements.
+
+The reason for not producing real evidence could be a wrong build of Attestation Agent for the TEE that you are attempting to use, or the Confidential VM not exposing the expected interfaces.
+
+### 2. Perform remote attestation
+
+Run `kubectl exec deploy/nginx -- curl http://127.0.0.1:8006/aa/token\?token_type\=kbs`. This should produce an attestation token. If you don't receive a token but an error you should inspect your Trustee KBS/AS logs and see whether there was a connection attempt for the CVM and potentially a reason why remote attestation failed. 
+
+You can set `RUST_LOG=debug` as environment variable the Trustee deployment to receive more verbose logs. If the evidence is being sent to KBS, the issue is most likely resolvable on the KBS side and possibly related to the evidence not meeting the expectations for a key release in KBS/AS policies or the requested resource not being available in the KBS.
+
+If you don't see an attestation attempt in the log of KBS there might be problems with network connectivity between the Confidential VM and the KBS. Note that the Pod and the Guest Components on the CVM might not share the same network namespace. A Pod might be able to reach KBS, but the Attestation Agent on the Confidential VM might not. If you are using HTTPS to reach the KBS, there might be a problem with the certificate provided to the Confidential VM (e.g via Initdata).
```

Comment:
I've seen a number of people using `127.0.0.1:8080` as the KBS URI. You sort of address this but maybe we should explicitly say not to do that. Also might be good to have an example of the annotation for the KBS URI because some people don't set it at all.

Reply:
where are people setting this? as part of the KID at encryption time or in their AA/CDH config?

Reply:
in their k8s annotation for the kernel params. I guess there are multiple places this can be set so maybe it's beyond the scope of this

Reply:
hmm, yeah. not sure what would motivate someone to configure their KBS to localhost, but maybe there is some test deployment scenario where this makes sense, so I'd say correct configuration of KBS uris is out-of-scope for this debug section about encrypted images.